lib/iex/lib/iex/info.ex

```diff
@@ -119,23 +119,51 @@ defimpl IEx.Info, for: List do
   def info(list) do
     specific_info =
       cond do
-        list == [] -> info_list(list)
-        List.ascii_printable?(list) -> info_charlist(list)
-        Keyword.keyword?(list) -> info_kw_list(list)
-        List.improper?(list) -> info_improper_list(list)
-        true -> info_list(list)
+        list == [] ->
+          info_list(list)
+
+        List.charlist?(list) ->
+          cond do
+            List.ascii_printable?(list) -> info_printable_charlist(list)
+            true -> info_non_printable_charlist(list)
+          end
+
+        Keyword.keyword?(list) ->
+          info_kw_list(list)
+
+        List.improper?(list) ->
+          info_improper_list(list)
+
+        true ->
+          info_list(list)
       end
 
     [{"Data type", "List"}] ++ specific_info
   end
 
-  defp info_charlist(charlist) do
+  defp info_printable_charlist(charlist) do
```

Comment:
I don't want to introduce the concept of a printable charlist. We don't print charlists, as far as I know.

Reply:
Actually, we do print charlists, but only if they are made only of ASCII characters. So I would rather keep the explanation we had before.